mqtt 3/1.0.0/modules/ROOT/pages/edits/mqtt3-connector-studio-configuration.adoc

```diff
@@ -0,0 +1,409 @@
+
+If the clean session is set to `True (Default)`, then when the connector disconnects, for whatever reason, all its subscriptions are dropped and the connector has to resubscribe upon reconnection. All messages sent while offline are lost.
+
+In the following example, you configure the clean session:
+
+. In Studio, navigate to the *Global Elements* tab.
+. Click *Create*.
+. In the filter box type `mqtt` and select *MQTT3 Config*.
+. Click *OK*.
+. In the *MQTT3 Config* window, for *Connection*, select *MQTT3 URL Connection*.
+. Set the following fields:
++
+* *Client ID*: `smart-bentley-123`.
+* *URL*: `tcp://127.0.0.1:1883`.
+* *Clean session*: `False`.
+[start=7]
+. Click *OK*.
+
+.MQTT3 Clean Session configuration
+image::mqtt3-cleansession.png[MQTT3 Clean session configuration in Global Element Properties window]
+
+In the XML editor, the `cleanSession` configuration looks like this:
+
+[source,xml,linenums]
+----
+<mqtt3:config name="MQTT_Config">
+  <mqtt3:connection clientId="smart-bentley-123" url="tcp://127.0.0.1:1883">
+      <mqtt3:connection-options cleanSession="false"/>
+    </mqtt3:connection>
+</mqtt3:config>
+----
+
+[[enable-file-persistence]]
+== Enable File Persistence
+
+In the *MQTT3 Config* window, set the *Enable file persistence* field to `True` to enable the MQTT client to persist its state to a file that is used to store any outbound or inbound in-flight messages that the client might have with QoS ≥ 1. In contrast, if you set the field to `False (Default)`, the client state is saved only in memory and in the event of a crash the client is not be able to recover its state.
+
+In the following example, you enable the file persistence:
+
+. In Studio, navigate to the *Global Elements* tab.
+. Click *Create*.
+. In the filter box type `mqtt` and select *MQTT3 Config*.
+. Click *OK*.
+. In the *MQTT3 Config* window, for *Connection*, select *MQTT3 URL Connection*.
+. Set the following fields:
++
+* *Client ID*: `smart-bentley-123`.
+* *Clean session*: `False`.
+* *Enable file persistence*: `True`.
+[start=7]
+. Set *Datastore* to `mqtt/store` to specify where you want the persistent store to be generated.
+. Click *OK*.
+
+.MQTT3 Enable File Persistence configuration
+image::mqtt3-enablefilepersistence.png[MQTT3 Enable File Persistence configuration in Global Element Properties window]
+
+In the XML editor, the `enableFilePersistence` and `dataStorePath` configurations look like this:
+
+[source,xml,linenums]
+----
+<mqtt3:config name="MQTT_Config">
+  <mqtt3:connection clientId="smart-bentley-123">
+       <mqtt3:connection-options cleanSession="false" enableFilePersistence="true" dataStorePath="mqtt/store"/>
+    </mqtt3:connection>
+</mqtt3:config>
+----
+
+[WARNING]
+If you set a dynamic client ID, MQTT 3 Connector will not be able to recover the persisted files in the event of a crash. You must set a client ID that will not change in the event of an application restart after a crash.
```

Comment:
```suggestion
If you set a dynamic client ID, MQTT 3 Connector cannot recover the persisted files in the event of a crash. You must set a client ID that does not change in the event of an application restart after a crash.
```
Can we use just one consistent wording for crash/restart/service interruption/etc.?